.github/workflows/hardware-smoke.yml

```
@@ -268,3 +268,61 @@ jobs:
           python3 ./scripts/dmc_rtt.py -n --openocd /opt/tenstorrent/bin/openocd-rtt
           echo "SMC RTT logs:"
           python3 ./scripts/smc_rtt.py -n --openocd /opt/tenstorrent/bin/openocd-rtt
+
+  build-fw-artifact:
+    uses: ./.github/workflows/build-fw.yml
+    secrets:
+      SIGNATURE_KEY: ${{ secrets.SIGNATURE_KEY }}
+  metal-test:
+    needs: build-fw-artifact
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - board: p100
+            runs-on:
+              - p100-jtag
+          - board: p100a
+            runs-on:
+              - p100a-jtag
+          - board: p150a
+            runs-on:
+              - p150a-jtag
+    runs-on: ${{ matrix.config.runs-on }}
+    container:
+      image: ghcr.io/tenstorrent/tt-metal/upstream-tests-bh:v0.58.0-rc26-5-g668476a4d2
+      volumes:
+        - /dev/hugepages-1G:/dev/hugepages-1G
+        - /dev/hugepages:/dev/hugepages
+      options: '--device /dev/tenstorrent --device /dev/bus/usb --privileged'
+    env:
+      ARCH_NAME: blackhole
+    defaults:
+      run:
+        shell: bash
+        working-directory: /home/user/tt-metal/
+    steps:
+      - name: Download the latest firmware bundle
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ needs.build-fw-artifact.outputs.combined-fwbundle-artifact }}
+          path: .
```

@kkabilarTT - this step seems to be failing consistently, and it appears that the directory probably does not exist in the container.

```shell
Run actions/download-artifact@v4
...
Starting download of artifact to: /__w/tt-zephyr-platforms/tt-zephyr-platforms
...
```
Can you try this instead?
```suggestion
    steps:
      - name: Download the latest firmware bundle
        uses: actions/download-artifact@v4
        with:
          name: ${{ needs.build-fw-artifact.outputs.combined-fwbundle-artifact }}
          path: /home/user/tt-metal
```

---

This is a great suggestion @cfriedt !

---

@kkabilarTT - I'm not 100% certain if `/home/user/tt-metal` already exists in the container, but if it does not, it would be necessary to create the directory in advance of using it.

So something like this above this rule.

```yaml
      - id: mkdir-home-user-tt-metal
        run: |
          mkdir -p /home/user/tt-metal
```